File: src/main/java/org/checkerframework/checker/mungo/typestate/graph/exceptions/ReservedStateName.java
package org.checkerframework.checker.mungo.typestate.graph.exceptions;

import org.checkerframework.checker.mungo.typestate.ast.TStateNode;

public class ReservedStateName extends TypestateError {

  public static final long serialVersionUID = 0L;

  public final TStateNode state;

  public ReservedStateName(TStateNode state) {
    this.state = state;
  }

  @Override
  public String getMessage() {
    return String.format("%s is a reserved state name (%s)", state.name, state.pos.toString());
  }

}

File: src/main/kotlin/org/checkerframework/checker/mungo/typestate/graph/exceptions/ReservedStateName.kt
package org.checkerframework.checker.mungo.typestate.graph.exceptions

import org.checkerframework.checker.mungo.typestate.ast.TStateNode

class ReservedStateName(val state: TStateNode) : TypestateError() {
  override val message: String
    get() = String.format("%s is a reserved state name (%s)", state.name, state.pos.toString())

  companion object {
    const val serialVersionUID = 0L
  }
}
